physical/raft/snapshot.go

```diff
@@ -17,30 +27,30 @@ const (
 	// boltSnapshotID is the stable ID for any boltDB snapshot. Keeping the ID
 	// stable means there is only ever one bolt snapshot in the system
 	boltSnapshotID = "bolt-snapshot"
+	tmpSuffix      = ".tmp"
+	snapPath       = "snapshots"
 )
 
-// BoltSnapshotStore implements the SnapshotStore interface and allows
-// snapshots to be made on the local disk. The main difference between this
-// store and the file store is we make the distinction between snapshots that
-// have been written by the FSM and by internal Raft operations. The former are
-// treated as noop snapshots on Persist and are read in full from the FSM on
-// Open. The latter are treated like normal file snapshots and are able to be
-// opened and applied as usual.
+// BoltSnapshotStore implements the SnapshotStore interface and allows snapshots
+// to be stored in BoltDB files on local disk. Since we always have an up to
+// date FSM we use a special snapshot ID to indicate that the snapshot can be
+// pulled from the BoltDB file that is currently backing the FSM. This allows us
+// to provide just-in-time snapshots without doing incremental data dumps.
+//
+// When a snapshot is being installed on the node we will Create and Write data
+// to it. This will cause the snapshot store to create a new BoltDB file and
+// write the snapshot data to it. Then, we can simply rename the snapshot to the
+// FSM's filename. This allows us to atomically install the snapshot and
+// reduces the amount of disk i/o.
```

Comment:
Maybe add a small line to the godoc that states if/when failed snapshots are deleted (WRT to being atomic).